internal/app/singularity/overlay_create.go

```diff
@@ -70,7 +71,32 @@ func addOverlayToImage(imagePath, overlayPath string) error {
 	return f.AddObject(di)
 }
 
-func OverlayCreate(size int, imgPath string, overlayDirs ...string) error {
+// findConvertCommand finds dd unless overlaySparse is true
+func findConvertCommand(overlaySparse bool) (string, error) {
+	// We can support additional arguments, so return a list
+	command := ""
+
+	// Sparse overlay requires truncate -s
+	if overlaySparse {
+		truncate, err := exec.LookPath(truncateBinary)
```

Comment:
`truncate` needs to be added to the `bin.FindBin` code. We now have *every* external binary we call go through this, so that the complete list of binaries is in a single place, and we can change path vs conf file lookup behavior centrally.

Reply:
> We now have every external binary we call go through this

To clarify - non-test code. There will be `exec.LookPath` for binaries within testing code still.

Reply:
oh that's why it didn't turn up as a result! I was wondering about that... :laughing:

Reply:
All set! https://github.com/sylabs/singularity/pull/864/commits/9eccb4ade29cf3ac7aef0e3a554d10315250887c